src/Illuminate/Support/Str.php

```diff
@@ -644,6 +644,24 @@ public static function substr($string, $start, $length = null)
         return mb_substr($string, $start, $length, 'UTF-8');
     }
 
+    /**
+     * Returns the number of substring occurrences.
+     *
+     * @param  string  $haystack
+     * @param  string  $needle
+     * @param  int  $offset
+     * @param  int|null  $length
+     * @return int
+     */
+    public static function substrCount($haystack, $needle, $offset = 0, $length = null)
+    {
+        if ($length) {
```

Comment:
This should be `! is_null($length)` is that `0` works properly.

Reply:
True. Fixed
Thanks!